packages/code-du-travail-frontend/src/common/Answer.js

```diff
@@ -195,6 +220,65 @@ const StyledTooltip = styled(Tooltip)`
   color: ${colors.lightText};
   border: solid 1px ${colors.elementBorder};
 `;
+
 const Underline = styled.span`
   border-bottom: 1px dotted ${colors.blueLight};
 `;
+
+function usePortals(children, html) {
+  const [portalComponents, setPortalComponents] = useState();
+  useEffect(() => {
+    const nodes = Array.from(
+      document.querySelectorAll(
+        "[data-main-content] p, [data-main-content] li:not([role=tab])"
+      )
+    ).reduce((state, node) => {
+      glossary.forEach(item => {
+        // we cannot use \b word boundary since \w does not match diacritics
+        // So we do a kind of \b equivalent.
+        // the main différence is that matched pattern can include a whitespace as first char
+        const frDiacritics = "àâäçéèêëïîôöùûüÿœæÀÂÄÇÉÈÊËÎÏÔÖÙÛÜŸŒÆ";
+        const wordBoundaryStart = `(?:^|[^\\w${frDiacritics}])`;
+        const wordBoundaryEnd = `(?![\\w${frDiacritics}])`;
+        const patterns = [...new Set([item.title, ...item.variants])]
+          .map(
+            term =>
+              new RegExp(`${wordBoundaryStart}${term}${wordBoundaryEnd}`, "gi")
+          )
+          .concat(item.abbrs.map(abbr => new RegExp(`\\b${abbr}\\b`, "g")));
+
+        patterns.forEach(pattern => {
+          node.innerHTML = node.innerHTML.replace(pattern, function(match) {
+            if (new RegExp(`^[^\\w${frDiacritics}]`).test(match)) {
+              // Since match string can start with a space, we trim it and insert the space before the tooltip markup
+              return `${match.slice(0, 1)}<span data-tooltip-slug="${
+                item.slug
+              }" data-tooltip-term="${match.slice(1)}"></span>`;
+            }
+            return `<span data-tooltip-slug="${item.slug}" data-tooltip-term="${match}"></span>`;
+          });
+        });
+      });
+
+      return state.concat(
+        Array.from(node.querySelectorAll("[data-tooltip-slug]")).map(node => ({
+          node,
+          term: node.getAttribute("data-tooltip-term"),
+          definition:
+            glossaryBySlug[node.getAttribute("data-tooltip-slug")].definition
+        }))
+      );
+    }, []);
+    setPortalComponents(
+      nodes.map(({ node, term, definition }, i) => {
+        return (
+          <Portal key={`item-${i}`} node={node}>
+            <DefinitonTerm term={term} definition={definition} />
+          </Portal>
+        );
+      })
+    );
+    return function cleanEffect() {};
+  }, [children, html]);
+  return portalComponents;
+}
```

Comment:
thanks for extracting; can this be exported to its own file ?

Reply:
and rename useGlossary() ;)

Reply:
Done